### PR TITLE
burn wounds, for real this time

### PR DESCRIPTION
**Pull request by hal9000PR**
## What Does This PR Do
the return of #15959
the concept is the same, the execution is however different. 
Right now, getting a burn wound will deal *permanent* damage until it is fixed. this damage is equal to the minimum breaking threshold for that limb. so the bigger the limb the harder it is to burn but the more max HP you lose when you do get burnt. eg the chest will remove 35 HP and a foot will lose 15 HP.

ointment, aloe and burn kits will prevent the wound getting infected for 3 minutes. after 3 minutes they will partially heal some of the "Permanent" damage. if it heals all the permanent damage the wound goes away. this will take 3 ointments on legs so 9 minutes or 4 on the chest so 12 minutes. 

Debridement surgery will fix them

walking around with a burn wound will make the infection worse, if its not ointmented.

I need to make nanocalcium and medbeams fix these. 

plasmamen and golems are immune as they have no skin

## Why It's Good For The Game
Lasers are still, almost always, a worse option than ballistics. nothing we do to ballistics will change that as long as they can break bones and cause IB. hopefully this PR will put ballistics and lasers on a more even playing field so there is actually a decent reason to use lasers that NT should be preferring according to lore.

## Testing
burnt myself, 
ointmented myself
surgered myself

## Changelog
:cl:
add: Burn wounds. Similar to broken bones or IB but caused by heat and burn damage. will permanently reduce that limbs HP and get infected until fixed with surgery or ointment and time.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human_damage.dm [hunk 1]
```diff
@@ -102,6 +98,9 @@
 		take_overall_damage(0, amount, updating_health, used_weapon = damage_source)
 	else
 		heal_overall_damage(0, -amount, updating_health, FALSE, robotic)
+
+	if(((maxHealth - getFireLoss()) < HEALTH_THRESHOLD_DEAD * 2) && stat == DEAD)
```
Comment (hal9000PR) [inline code/modules/mob/living/carbon/human/human_damage.dm:102]: I need to test this works

File: code/modules/surgery/organs/organ_external.dm [hunk 2]
```diff
@@ -361,10 +363,10 @@ This function completely restores a damaged organ to perfect condition.
 				trace_chemicals[chemID] = trace_chemicals[chemID] - 1
 				if(trace_chemicals[chemID] <= 0)
 					trace_chemicals.Remove(chemID)
-
+/*
 		if(!(status & ORGAN_BROKEN))
 			perma_injury = 0
-
+*/
```
Comment (hal9000PR) [inline code/modules/surgery/organs/organ_external.dm:369]: This needs removing

File: code/modules/mob/living/carbon/human/human_life.dm [hunk 3]
```diff
@@ -686,7 +686,7 @@
 				// Use the dead health doll as the base, since we have proper "healthy" overlays now
 				healthdoll.icon_state = "healthdoll_DEAD"
 				for(var/obj/item/organ/external/O in bodyparts)
-					var/damage = O.burn_dam + O.brute_dam
+					var/damage = O.get_damage()
```
Comment (hal9000PR) [inline code/modules/mob/living/carbon/human/human_life.dm:689]: This needs to show up on examine

File: code/modules/surgery/organs/organ_external.dm [hunk 4]
```diff
@@ -655,7 +661,6 @@ Note that amputating the affected organ does in fact remove the infection from t
 
 	status |= ORGAN_BROKEN
 	broken_description = pick("broken", "fracture", "hairline fracture")
-	perma_injury = brute_dam
```
Comment (hal9000PR) [inline code/modules/surgery/organs/organ_external.dm:658]: This never worked

File: code/modules/surgery/organs/organ_external.dm [hunk 5]
```diff
@@ -687,7 +691,16 @@ Note that amputating the affected organ does in fact remove the infection from t
 	if(is_robotic())
 		return
 	status &= ~ORGAN_INT_BLEEDING
-	perma_injury = 0
```
Comment (hal9000PR) [inline code/modules/surgery/organs/organ_external.dm:690]: This was a copy paste error by me 15 months ago

**Comment by hal9000PR** [comment]
update: bodyscanner feedback
image
self examine feedback
image
regular examine feedback
image

File: code/modules/mob/living/carbon/human/human_damage.dm (hunk 1 above)
Comment (hal9000PR) [reply code/modules/mob/living/carbon/human/human_damage.dm:102]: tested, it works

File: code/game/objects/items/stacks/medical_packs.dm [hunk 6]
```diff
@@ -214,6 +215,21 @@
...
 			to_chat(user, "<span class='warning'>[affecting] is cut open, you'll need more than some ointment!</span>")
 
+/obj/item/stack/medical/ointment/heal(mob/living/M, mob/user)
+	var/obj/item/organ/external/affecting = ..()
+	if((affecting.status & ORGAN_BURNT) && !(affecting.status & ORGAN_SALVED))
+		affecting.status |= ORGAN_SALVED
+		addtimer(CALLBACK(affecting, TYPE_PROC_REF(/obj/item/organ/external, remove_ointment), heal_burn), 3 MINUTES)
+
+
+
```
Comment (S34NW) [inline code/game/objects/items/stacks/medical_packs.dm:225]: too much whitespace

File: code/modules/mob/living/carbon/carbon.dm [hunk 7]
```diff
@@ -318,20 +318,25 @@
...
+				if(10 to 40)
+					status += "blistered"
+				if(40 to INFINITY)
+					status += "peeling away"
+
 
 		if(LB.status & ORGAN_MUTATED)
 			status = "weirdly shapen"
 
+
```
Comment (S34NW) [inline code/modules/mob/living/carbon/carbon.dm:339]:
```suggestion
```

File: code/modules/mob/living/carbon/human/human_organs.dm [hunk 8]
```diff
@@ -25,6 +25,10 @@
 				var/obj/item/organ/internal/I = pick(E.internal_organs)
 				custom_pain("You feel broken bones moving in your [E.name]!")
 				I.receive_damage(rand(3,5))
+			if((E.status & ORGAN_BURNT) && !(E.status & ORGAN_SALVED))
+				custom_pain("You rub the burn wound on [E.name]!")
```
Comment (S34NW) [inline code/modules/mob/living/carbon/human/human_organs.dm:29]:
```suggestion
				custom_pain("You feel the skin falling off the burn wound on your [E.name]!")
```

File: code/modules/surgery/organs/organ_external.dm [hunk 9]
```diff
@@ -447,6 +448,10 @@ Note that amputating the affected organ does in fact remove the infection from t
 	if(damage > 15 && local_damage > 30 && prob(damage))
 		cause_internal_bleeding()
 
+/obj/item/organ/external/proc/check_for_burn_wound(damage)
+	if(burn_dam > min_broken_damage && prob(damage * max(owner.bodytemperature / BODYTEMP_HEAT_DAMAGE_LIMIT, 1)))
```
Comment (S34NW) [inline code/modules/surgery/organs/organ_external.dm:452]:
```suggestion
	if(burn_dam >= min_broken_damage && prob(damage * max(owner.bodytemperature / BODYTEMP_HEAT_DAMAGE_LIMIT, 1)))
```

**Review by S34NW — COMMENTED** [review]
_Extremely_ minor points before this is good to go. Amazing work.

**Review by S34NW — APPROVED** [review]
Think it should be good now

File: code/modules/surgery/organs/organ.dm [hunk 10]
```diff
@@ -104,7 +104,7 @@
 		germ_level = 0
 		return
 
-	if(!owner)
+	if(!owner || (status & ORGAN_BURNT && !(status & ORGAN_SALVED)))
```
Comment (farie82) [inline code/modules/surgery/organs/organ.dm:107]:
```suggestion
	if(!owner || ((status & ORGAN_BURNT) && !(status & ORGAN_SALVED)))
```

File: code/modules/surgery/organs/organ_external.dm [hunk 11]
```diff
@@ -689,7 +692,18 @@ Note that amputating the affected organ does in fact remove the infection from t
 	if(is_robotic())
 		return
 	status &= ~ORGAN_INT_BLEEDING
-	perma_injury = 0
+
+/obj/item/organ/external/proc/cause_burn_wound()
+	if(is_robotic() || limb_flags & CANNOT_BURN || status & ORGAN_BURNT)
```
Comment (farie82) [inline code/modules/surgery/organs/organ_external.dm:697]:
```suggestion
	if(is_robotic() || (limb_flags & CANNOT_BURN) || (status & ORGAN_BURNT))
```

**Review by farie82 — COMMENTED** [review]
Two minor styling things

**Review by farie82 — APPROVED** [review]

File: code/modules/surgery/organs/organ_external.dm [hunk 12]
```diff
@@ -447,6 +448,10 @@ Note that amputating the affected organ does in fact remove the infection from t
 	if(damage > 15 && local_damage > 30 && prob(damage))
 		cause_internal_bleeding()
 
+/obj/item/organ/external/proc/check_for_burn_wound(damage)
+	if(burn_dam >= min_broken_damage && prob(damage * max(owner.bodytemperature / BODYTEMP_HEAT_DAMAGE_LIMIT, 1)))
```
Comment (GDNgit) [inline code/modules/surgery/organs/organ_external.dm:452]: Owner needs to be sanity checked here, an IPC who gets a limb blow off won't have one
[2023-03-30T12:10:56] Runtime in organ_external.dm,452: Cannot read null.bodytemperature

File: code/game/objects/items/devices/scanners.dm [hunk 13]
```diff
@@ -213,20 +215,18 @@ REAGENT SCANNER
...
 		if(e.has_infected_wound())
 			to_chat(user, "<span class='warning'>Infected wound detected in subject [limb]. Disinfection recommended.</span>")
+		burn_wound = burn_wound || (e.status & ORGAN_BURNT)
+		internal_bleed = internal_bleed || (e.status & ORGAN_INT_BLEEDING)
+	if(broken_bone)
+		to_chat(user, "<span class='warning'>Bone fractures detected. Advanced scanner required for location.</span>")
+	if(internal_bleed)
+		to_chat(user, "<span class='warning'>Internal bleeding detected. Advanced scanner required for location.</span>")
+	if(burn_wound)
+		to_chat(user, "<span class='warning'>Critical Burn detected. Examine patients body for location.</span>")
```
Comment (CinnamonSnowball) [inline code/game/objects/items/devices/scanners.dm:228]:
```suggestion
		to_chat(user, "<span class='warning'>Critical burn detected. Examine patient's body for location.</span>")
```

**Review by CinnamonSnowball — COMMENTED** [review]
It looks like critical burns are missing from Body Scanner printouts
Also, minor spelling stuff

**Comment by pismorizzo** [comment]
Any shot cling's flesh mend can fix burn wounds too, just doesn't make sense to me that flesh mend would be unable to fix necrotized tissue but a dab of ointment could.

**Comment by hal9000PR** [comment]
> Any shot that cling's flesh mend can be allowed to fix burn wounds too, would make sense that a ling that can mend it's own flesh and restore incredible damage automatously could fix just some burned tissue.

Maybe in a future pr. Not right now the scope of this pr is already dangerous with touching species.

**Review by S34NW — APPROVED** [review]

File: code/game/machinery/adv_med.dm [hunk 14]
```diff
@@ -464,6 +466,10 @@
 				splint = "Splinted:"
 			if(e.status & ORGAN_BROKEN)
 				AN = "[e.broken_description]:"
+			if(e.status & ORGAN_SALVED)
+				ointment = "Salved:"
+			if(e.status & ORGAN_BURNT)
+				burn_wound= "Critical Burn:"
```
Comment (farie82) [inline code/game/machinery/adv_med.dm:472]:
```suggestion
				burn_wound = "Critical Burn:"
```

**Review by farie82 — APPROVED** [review]

**Pull request merged**